pkg/projectfile/projectfile.go

```diff
@@ -606,15 +607,37 @@ func (p *Project) save(path string) *failures.Failure {
 	return nil
 }
 
+func (p *Project) SetNamespaceAndCommit(owner, project, commitID string) error {
```

Comment:
This is doing too much. Just change this to `SetNamespace`, then cal `SetCommit` separately. We're not trying to shave off milliseconds of IO.